doc/SCC-API-RFC-(Draft).md

```diff
@@ -0,0 +1,372 @@
+                "target distribution": "sles-12-x86_64",
+                "tags": [
+                  "enabled",
+                  "autorefresh"
+                ]
+              }
+            ]
+          }
+        ]
+      }
+    ]
+  }
+]
+```
+[^](#systems)
+***
+
+##### <a id="sys_prod_list">List Products</a>
+List all products for the system. This includes a list of all repositories
+for each product.
+```
+GET /connect/systems/products
+```
+###### Parameters
+  - optional:
+    - `product_id` ( *String* ): Zypper CPE product identifier
+    - `type` ( *String* ): Type of product. Valid values: `extension`, `base`. [Not implemented]
+
+###### Response
+```
+Status: 200 OK
+```
+```json
+[
+
+    {
+        "id": 293,
+        "name": "Extension",
+        "created_at": "2014-03-07T12:55:54.829Z",
+        "updated_at": "2014-03-07T12:55:54.829Z",
+        "productdataid": 77238,
+        "nnw_product_data": "ab12cdff6532",
+        "edition_id": 37,
+        "architecture_id": 12,
+        "zypper_name": "Zypper Product Name 1",
+        "release_type": null
+    }
+
+]
+```
+[^](#systems)
+***
+
+
+##### <a id="sys_service_list">List Services</a>
+List all services for the system.
+```
+GET /connect/systems/services
+```
+###### Parameters
+  - optional:
+    - `product_id` ( *Integer* ): Product ID.
+
+###### Response
+```
+Status: 200 OK
+```
+```json
+[
+  {
+    "repositories": [
+                      {
+                      "name": "SLE12-SP1-Pool",
+                      "archs": [
+                        "x86_64"
+                       ],
+                      "url": "https://download.suse.com/repo/SLE12-SP1-Pool/sles-12-x86_64",
+                      "target distribution": "sles-12-x86_64",
+                      "tags": [
+                        "enabled",
+                        "autorefresh"
+                      ]
+                     },
+                    ...
+                    ]
+  }
+]
+```
+[^](#systems)
+***
+
+
+##### <a id="sys_repo_list">List Repositories</a>
+List all repositories for the system.
+```
+GET /connect/systems/repos
+```
+###### Parameters
+  - optional:
+    - `product_id` ( *Integer* ): Product ID.
+
+###### Response
+```
+Status: 200 OK
+```
+```json
+[
+  {
+    "name": "SLE12-SP1-Pool",
+    "archs": [
+      "x86_64"
+     ],
+    "url": "https://download.suse.com/repo/SLE12-SP1-Pool/sles-12-x86_64",
+    "target distribution": "sles-12-x86_64",
+    "tags": [
+      "enabled",
+      "autorefresh"
+    ]
+  }
+]
+```
+[^](#systems)
+***
+***
+
+
+### <a id="public">Public</a>
+For all endpoints on this level there is no authentication needed. They are
+publicly available. Of course, that means that only publicly available resources
+will be listed.
+
+The typical use case for these calls is the SLES installer
+
+  - [list products](#prod_list)
+  - [list services](#service_list)
+  - [list repositories](#repo_list)
```

Comment:
Draft we can ignore as that was just a starting point back then. We will keep it for the history, but there is no point in fixing it from my POV.